src/main/scala/com/codacy/analysis/cli/files/GitFileCollector.scala

```diff
@@ -0,0 +1,56 @@
+package com.codacy.analysis.cli.files
+
+import java.nio.file.{Path, Paths}
+
+import better.files.File
+import com.codacy.analysis.cli.clients.api.ProjectConfiguration
+import com.codacy.analysis.cli.configuration.CodacyConfigurationFile
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import org.eclipse.jgit.treewalk.TreeWalk
+
+import scala.util.Try
+
+class GitFileCollector extends FileCollector[Try] {
+  override def list(directory: File,
+                    localConfiguration: Either[String, CodacyConfigurationFile],
+                    remoteConfiguration: Either[String, ProjectConfiguration]): Try[FilesTarget] = {
+
+    Try {
+      val builder = new FileRepositoryBuilder
```

Comment:
Maybe wrap the git logic in its own class and not in the collector.

Reply:
I know it's mutable code, but I think this is in the right place as this is the exact concern of the this method. What I could do is move all the jgit logic into a private function, but that would be this exact logic minus the two last lines, so it doesn't seem worth it.